numpy/ma/core.py

```diff
@@ -1296,14 +1296,23 @@ def _recursive_make_descr(datatype, newtype=bool_):
                 # Prepend the title to the name
                 name = (field[-1], name)
             descr.append((name, _recursive_make_descr(field[0], newtype)))
-        return descr
+        newtype = np.dtype(descr)
```

Comment:
Maybe instead of calling this `newtype` (which is confusing as it is also a passed-in argument), just keep calling it `descr`?